packages/atlas/src/views/viewer/VideoView/CommentsSection.tsx

```diff
@@ -226,7 +267,27 @@ export const CommentsSection: React.FC<CommentsSectionProps> = ({ disabled, vide
       }
     }
 
-    return comments?.map((comment, idx) =>
+    const filteredParentCommentReplies = parentCommentFromUrl
+      ? {
+          ...parentCommentFromUrl,
+          replies: parentCommentFromUrl?.replies?.filter((comment) => comment.id !== commentIdQueryParam) || [],
+        }
+      : null
+
+    const filteredComments = commentFromUrl
+      ? [
+          ...(commentFromUrl.parentCommentId
+            ? filteredParentCommentReplies
+              ? [filteredParentCommentReplies, commentFromUrl]
+              : []
+            : [commentFromUrl]),
+          ...(comments?.filter(
+            (comment) => ![commentFromUrl.id, commentFromUrl.parentCommentId].includes(comment.id)
+          ) || []),
+        ]
+      : comments
```

Comment:
This is super hard to read, I have no idea what's going on. Maybe we could:
1. Split this into more intermediary steps with named variables or use `if/else`? Not sure, but what we have currently doesn't work great 
2. Extract part of this comments logic to some local hook to encapsulate? (just an idea, up to you)